feat: command menu v2 ui components

charislam opened the pull request:
Add all the UI components needed for Command Menu v2, and document it on the design system site.

charislam committed:
feat: finish command menu base

charislam committed:
add command menu v2 to design system site

charislam committed:
format

charislam committed:
add example of commands changing in response to state changes

MildTomato commented:
To maybe improve this, maybe support for a Trigger component, so that all the right aria labels are used. See the [Dialog component](http://localhost:3003/design-system/docs/components/dialog)

<img width="1670" alt="Screenshot 2024-07-11 at 3 44 14 PM" src="https://github.com/supabase/supabase/assets/8291514/37a02cb4-2bd7-4328-85b2-4e601f49090c">

MildTomato reviewed (approved):
Approved.

No changes to apps directly from what I can see.

charislam committed:
fix: add trigger with proper a11y

charislam committed:
feat: add mobile ux improvements from main

charislam committed:
fix: mobile positioning

fsansalvadore committed:
Merge branch 'master' into feat/command-menu-v2-pt-ivelosttrack

fsansalvadore commented:
The cmdk looked a bit funky and docs had old nav.
I resolved a small conflict and now it's up to date.

fsansalvadore commented on packages/common/hooks/index.ts:
```diff
@@ -8,3 +8,4 @@ export * from './useParams'
 export * from './useSearchParamsShallow'
 export * from './useTelemetryProps'
 export * from './useThemeSandbox'
+export * from './useDragToClose'
```
```suggestion
```

fsansalvadore replied:
duplicate

charislam committed:
test: swap jest for vitest

Once again experiencing problems with imports for Jest, and the quickest
way to solve is to swap for Vitest.

charislam committed:
refactor: remove unnecessary css stubs (vitest doesn't need them)
